### PR TITLE
feat: adds unit summaries to listings list to backend

### PR DESCRIPTION
**Pull request by seanmalbert**
## Issue

- Addresses #1059 in part

## Description

This adds unit summaries to the listings list for use on the frontend browse and filtered pages.

## How Can This Be Tested/Reviewed?

Reseed the db locally and see that unitSummaries are on the listings.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/src/listings/listings.service.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export class ListingsService {
     @Inject(REQUEST) private req: ExpressRequest,
     @InjectRepository(Listing) private readonly listingRepository: Repository<Listing>,
     @InjectRepository(AmiChart) private readonly amiChartsRepository: Repository<AmiChart>,
+    @InjectRepository(UnitGroup) private readonly unitGrouopRepository: Repository<UnitGroup>,
```
Comment (pbn4) [inline backend/core/src/listings/listings.service.ts:42]: There is a typo in `Grouop`

File: backend/core/src/shared/units-transformations.ts [hunk 2]
```diff
@@ -12,15 +12,10 @@ import { MonthlyRentDeterminationType } from "../units-summary/types/monthly-ren
...
-// One row for every unit group with occupancy / sq ft / floor range averaged across all unit types
-// Used to display the occupancy table, unit type accordions
-export const getUnitTypeSummary = (unitGroups: UnitGroup[]): UnitTypeSummary[] => {
-  return []
-}
-
 // One row for every unit group, with rent and ami ranges across all ami levels
 // Used to display the main pricing table
 export const getUnitGroupSummary = (unitGroups: UnitGroup[] = []): UnitGroupSummary[] => {
+  console.log("unitGroups = ", unitGroups)
```
Comment (pbn4) [inline backend/core/src/shared/units-transformations.ts:18]: Can be removed

File: backend/core/src/listings/listings.service.ts [hunk 3]
```diff
@@ -153,6 +156,34 @@ export class ListingsService {
...
+    const res = await this.unitGrouopRepository.find({
+      cache: true,
+      where: {
+        listing: {
+          id: In(listings.map((listing) => listing.id)),
+        },
+      },
+    })
+
+    const unitGroupMap = res.reduce((obj, current) => {
```
Comment (pbn4) [inline backend/core/src/listings/listings.service.ts:169]: Could use typing `Record<string, Array<UnitGroup>>`

File: backend/core/src/units-summary/entities/unit-group.entity.ts [hunk 4]
```diff
@@ -42,6 +43,12 @@ export class UnitGroup {
   @ManyToOne(() => Listing, (listing) => listing.unitGroups, {})
   listing: Listing
 
+  @RelationId((unitGroupEntity: UnitGroup) => unitGroupEntity.listing)
+  @Expose()
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
```
Comment (pbn4) [inline backend/core/src/units-summary/entities/unit-group.entity.ts:48]: I think it's not optional, is it?

**Review by pbn4 — CHANGES_REQUESTED** [review]

**Comment by seanmalbert** [comment]
@pbn4 ,
I've addressed your changes. There are also a couple of unrelated updates, cleaning assets for seeds and a fix on the public side I noticed.

**Review by pbn4 — APPROVED** [review]

**Pull request merged**